pypesto/visualize/sampling.py

```diff
@@ -91,6 +114,565 @@ def sampling_fval_trace(
+        n_subplots = len(condition_ids)
+    elif groupby == OBSERVABLE:
+        n_variables = len(condition_ids)
+        y_names = condition_ids
+        n_subplots = len(observable_ids)
+    else:
+        raise ValueError(f'Unsupported groupby value: {groupby}')
+
+    levels_opacity = sorted(
+        # min 30%, max 100%, opacity
+        np.linspace(0.3 * RGBA_MAX, RGBA_MAX, len(levels)),
+        reverse=True,
+    )
+    if axes is not None:
+        fig = axes.get_figure()
+        if not isinstance(axes, np.ndarray):
+            axes = np.array([[axes]])
+    cmap = plt.cm.viridis
+    cmap_min = RGBA_MIN
+    cmap_max = 0.85*(RGBA_MAX - RGBA_MIN) + RGBA_MIN  # exclude yellows
+
+    # define colormap
+    variables_color = [
+        list(cmap(v))[:LEN_RGB]
+        for v in np.linspace(cmap_min, cmap_max, n_variables)
+    ]
+
+    if axes is None:
+        n_row = int(np.round(np.sqrt(n_subplots)))
+        n_col = int(np.ceil(n_subplots / n_row))
+        fig, axes = plt.subplots(n_row, n_col, figsize=size, squeeze=False)
+    elif len(axes.flat) < n_subplots:
+        raise ValueError(
+            'Provided `ax` contains insufficient subplots. At least '
+            f'{n_subplots} are required.'
+        )
+
+    if groupby == CONDITION:
+        _plot_trajectories_by_condition(
+            summary=summary,
+            condition_ids=condition_ids,
+            observable_ids=observable_ids,
+            axes=axes,
+            levels=levels,
+            levels_opacity=levels_opacity,
+            labels=labels,
+            variables_color=variables_color,
+        )
+    elif groupby == OBSERVABLE:
+        _plot_trajectories_by_observable(
+            summary=summary,
+            condition_ids=condition_ids,
+            observable_ids=observable_ids,
+            axes=axes,
+            levels=levels,
+            levels_opacity=levels_opacity,
+            labels=labels,
+            variables_color=variables_color,
+        )
+
+    if title:
+        fig.suptitle(title)
+
+    # Fake plots for legend line styles
+    fake_data = [[0], [0]]
+    lines_variables = np.array([
+        # Assumes that the color for a variable is always the same, with
+        # different opacity for different confidence interval levels.
+        [
+            labels[y_name],
+            Line2D(*fake_data, color=variables_color[index], lw=4)
+        ]
+        for index, y_name in enumerate(y_names)
+    ])
+    lines_levels = np.array([
+        # Assumes that different CI levels are represented as
+        # different opacities of the same color.
+        [
+            f'{level}% CI',
+            Line2D(
+                *fake_data,
+                color=rgba2rgb([
+                    *RGBA_BLACK[:LEN_RGB],
+                    levels_opacity[index]
+                ]),
+                lw=4
+            )
+        ]
+        for index, level in enumerate(levels)
+    ] + [['Median', Line2D(*fake_data, color=RGBA_BLACK)]]
+    )
```

Comment:
not sure, and matter of style... but feels weird to read a 17 line list comprehension...
Either maybe slighty rearranging, or may putting this into an actual loop (we got 4 levels of indentation here anyway) and putting it into a subfunction together with lines_variables?